kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java

```diff
@@ -225,6 +225,21 @@ public String userReference(String id)
 		return getAccessPoint(true) + Entity.SEPARATOR + ((id == null) ? "" : id);
 	}
 
+	public String idFromReference(String reference) {
+
+		if (StringUtils.isBlank(reference)) {
```

Comment:
Would add a check to make sure it's a user reference
if (StringUtils.isBlank(reference) || !reference.startsWith("/user/"))